apps/console/src/features/roles/components/wizard-updated/role-permissions/role-permissions.tsx

```diff
@@ -0,0 +1,276 @@
+                            "notifications.fetchAPIResourceError.error.description"),
+                        level: AlertLevels.ERROR,
+                        message: t("console:manage.features.roles.addRoleWizard.forms.rolePermission." +
+                            "notifications.fetchAPIResourceError.error.message")
+                    })
+                );
+                setIsPermissionStepNextButtonDisabled(true);
+            } else {
+                setIsPermissionStepNextButtonDisabled(false);
+            }
+        }, [ selectedAPIResourceFetchRequestError, apiResourcsListFetchRequestError ]);
+
+        useEffect(() => {
+            const options: DropdownProps[] = [];
+    
+            apiResourcesList?.apiResources?.map((apiResource: APIResourceInterface) => {
+                if (!selectedAPIResources.find((selectedAPIResource: APIResourceInterface) => 
+                    selectedAPIResource?.id === apiResource?.id)) {
+                    options.push({
+                        key: apiResource.id,
+                        text: apiResource.name,
+                        value: apiResource.id
+                    });
+                }
+            });
+        
+            setAPIResourcesListOptions(options);
+        }, [ apiResourcesList, selectedAPIResources ]);
+
+        /**
+         * Add API resource to the selected API resources list.
+         */
+        useEffect(() => {
+            if (selectedAPIResource 
+                && !isSelectedAPIResourceFetchRequestLoading 
+                && !isSelectedAPIResourceFetchRequestValidating) {
+                
+                if (!selectedAPIResources.find(
+                    (apiResource: APIResourceInterface) => selectedAPIResource?.id === apiResource?.id)) {
+                    setSelectedAPIResources([ selectedAPIResource, ...selectedAPIResources ]);
+                }
+
+                selectedAPIResourceId && setSelectedAPIResourceId(undefined);
+            }
+        }, [ selectedAPIResource ]);
+
+        /**
+         * The following function handles the search query for the groups list.
+         */
+        const searchAPIResources: DebouncedFunc<(query: string) => void> = 
+            useCallback(debounce((query: string) => {
+                setAPIResourceSearchQuery(`name co ${query}`);
+                mutateAPIResourcesListFetchRequest().finally(() => {
+                    setAPIResourcesSearching(false);
+                    setAPIResourceSearchQuery(undefined);
+                });
+            }, RoleConstants.DEBOUNCE_TIMEOUT), []);
+
+        /**
+         * Handles the change of the search query of application list.
+         */
+        const onSearchChangeAPIResourcs = (event: SyntheticEvent<HTMLElement>, data: DropdownProps): void => {
+            setAPIResourcesSearching(true);
+            searchAPIResources(data.searchQuery.toString().trim());
```

Comment:
Any chance of `searchQuery` being null or undefined?

Reply:
Fixed with https://github.com/wso2/identity-apps/pull/4278/commits/445e6e01208a111c7632813406802eb382873a7f